### PR TITLE
Added terra-props-table-loader

### LINKED ISSUE #199: Add terra-props-table-loader
# Feature Request

### PR DESCRIPTION
**Pull request by bjankord**
### Summary
Adds custom react-docgen based props-table loader so we can generate the props table doc at build time instead of compile time.

This correctly merges this addition with the [resolve loader we have defined in terra-toolkit](https://github.com/cerner/terra-toolkit/blob/master/config/webpack/webpack.config.js) resulting in the following:

```
  resolveLoader:
   { modules: [ '/Users/bj031910/repos/terra-core/node_modules' ],
     alias:
      { 'terra-props-table-loader':
         '/Users/bj031910/repos/terra-core/node_modules/terra-dev-site/config/webpack/loaders/terra-props-table-loader' } },
```

Resolves #199 

### Related PR
Here is a PR in terra-core that consumes this change to terra-dev-site:
* https://github.com/cerner/terra-core/pull/2642
* https://github.com/cerner/terra-clinical/pull/566

@cerner/terra

### CODE REVIEW

**Comment by mjhenkes** [comment]
For MDX, it would be nice if this returned a React component. What would you think of changing that now?

**Comment by bjankord** [comment]
> For MDX, it would be nice if this returned a React component. What would you think of changing that now?

I can take a look into it.

**Comment by mjhenkes** [comment]
Don't worry about changing the output to a react component. I figured out a clever way to chain these loaders together with a resource query
https://webpack.js.org/configuration/module/#ruleresourcequery

**Comment by mjhenkes** [comment]
Basically the user would change the import from
```
import TestFilePropsTable from '!terra-props-table-loader!../../../src/terra-dev-site/test/testfile.jsx';
```
To in mdx
```
import TestFilePropsTable from '../../../src/terra-dev-site/test/testfile.jsx?terra-props-table-loader';
```

File: tests/wdio/terra-props-table-loader-spec.js [hunk 1]
```diff
@@ -0,0 +1,6 @@
+Terra.describeViewports('terra-props-table-loader', ['huge'], () => {
+  it('checks that terra-props-table-loader works', () => {
+    browser.url('/raw/terra-props-table-loader/terra-dev-site/terra-props-table-loader');
+    Terra.validates.element();
```
[the remark this answers is not in the document]
Comment (bjankord) [reply tests/wdio/terra-props-table-loader-spec.js]: @mjhenkes fixed up the tests in this PR.

File: config/webpack/loaders/terra-props-table-loader/index.js [hunk 2]
```diff
@@ -0,0 +1,119 @@
+/* eslint-disable import/no-extraneous-dependencies */
```
[the remark this answers is not in the document]
Comment (bjankord) [reply config/webpack/loaders/terra-props-table-loader/index.js]: Added in this commit: https://github.com/cerner/terra-dev-site/pull/201/commits/992e0e05444b56902c9e8e613e66772a15759e47?short_path=d943787#diff-d943787f5f8088dd6b8c656adcab96b9

**Comment by mjhenkes** [comment]
We're going to close this PR and include it with the MDX PR

**Pull request closed without merging**